orthoinference/src/main/java/org/reactome/orthoinference/InferEWAS.java

```diff
@@ -0,0 +1,384 @@
+	private static GKInstance speciesInst;
+	private static HashMap<String, String[]> homologueMappings = new HashMap<String,String[]>();
+	private static HashMap<String, ArrayList<String>> ensgMappings = new HashMap<String,ArrayList<String>>();
+	private static HashMap<String, GKInstance> seenRGP = new HashMap<String,GKInstance>();
+	private static HashMap<String,GKInstance> ewasIdenticals = new HashMap<String,GKInstance>();
+	private static HashMap<String,GKInstance> residueIdenticals = new HashMap<String,GKInstance>();
+
+    // Creates an array of inferred EWAS instances from the homologue mappings file (hsap_species_mapping.txt).
+	@SuppressWarnings("unchecked")
+	public static ArrayList<GKInstance> ewasInferrer(GKInstance ewasInst) throws InvalidAttributeException, Exception
+	{
+		ArrayList<GKInstance> infEWASInstances = new ArrayList<GKInstance>();
+		String referenceEntityId = ((GKInstance) ewasInst.getAttributeValue(ReactomeJavaConstants.referenceEntity)).getAttributeValue(ReactomeJavaConstants.identifier).toString();
+		if (homologueMappings.get(referenceEntityId) != null)
+			{
+				// Iterate through the array of homologue mappings, attempting to infer EWAS instances for each.
+				for (Object homologue : homologueMappings.get(referenceEntityId))
+				{
+					String[] splitHomologue = homologue.toString().split(":");
+					String homologueSource = splitHomologue[0];
+					String homologueId = splitHomologue[1];
+					
+					GKInstance infReferenceGeneProduct = null;
+					if (seenRGP.get(homologueId) == null)
+					{
+						infReferenceGeneProduct = InstanceUtilities.newInferredGKInstance((GKInstance) ewasInst.getAttributeValue(ReactomeJavaConstants.referenceEntity));
+						infReferenceGeneProduct.addAttributeValue(ReactomeJavaConstants.identifier, homologueId);
+						// Reference DB can differ between homologue mappings, but can be differentiated by the 'homologueSource' found in each mapping.
+						GKInstance referenceDb = null;
+						if (homologueSource.equals("ENSP"))
+						{
+							referenceDb = enspDbInst;
+						} else {
+							referenceDb = uniprotDbInst;
+						}
+						infReferenceGeneProduct.addAttributeValue(ReactomeJavaConstants.referenceDatabase,  referenceDb);
+						
+						//// Creates ReferenceDNASequence instance from ReferenceEntity
+						ArrayList<GKInstance> inferredReferenceDNAInstances = InferEWAS.createReferenceDNASequence(homologueId);
+						infReferenceGeneProduct.addAttributeValue(ReactomeJavaConstants.referenceGene, inferredReferenceDNAInstances);
+						
+						infReferenceGeneProduct.addAttributeValue(ReactomeJavaConstants.species, speciesInst);
+						infReferenceGeneProduct.setAttributeValue(ReactomeJavaConstants._displayName, "UniProt:" + homologueId);
+						infReferenceGeneProduct = InstanceUtilities.checkForIdenticalInstances(infReferenceGeneProduct);
+						seenRGP.put(homologueId, infReferenceGeneProduct);
+					} else {
+						infReferenceGeneProduct = seenRGP.get(homologueId);
+					}
```

Comment:
Obtaining the inferred reference gene product makes sense to be extracted as its own private method